src/Bot.hs

```diff
@@ -25,8 +25,8 @@ app' = do
 
 typeRequest :: Text -> S.ActionM ()
 typeRequest func = do
-  hoogleRes <- liftIO . simpleHttp . functionNameToUrl $ removeCommandChars func
-  maybe badRequest (liftIO . slackRequest) (firstType =<< decode hoogleRes)
+  result <- hoogle func
+  maybe badRequest slackRequest result
```

Comment:
Nice. I think you could do `typeRequest = maybe badRequest slackRequest <=< hoogle` if you really want to shorten it up.